### DISCUSSION TITLE
Secure Ownership Signatures

### DISCUSSION THREAD

**Discussion by hanniabu** [WalletConnect/walletconnect-monorepo#1919] [Ideas]
Many apps ask you to sign a message to prove your ownership of the wallet. This is dangerous because as a user you have no idea what you're signing. It'd be great to develop a standard that all wallets can follow to provide proof of ownership as a first-class citizen. For example they call a specific function rather than creating their own message to sign, and in the signing UI it will indicate that this is to prove ownership of the wallet, and the only way we're presented with this message is if it's done through the wallet. This way users aren't signing random messages hoping we're not actually signing a transaction without knowing.

**Comment by GrahamMcBain** [comment]
These types of topics are actively being discussed by groups like CASA. I highly suggest watching the WalletCon Livestream or attending IRL if you can on March 1st!

**Reply by hanniabu** [reply]
If you're talking about https://keys.casa/, they don't even support Ethereum
